### PR TITLE
Add log/map- server health check

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

**Comment by Martin2112** [comment]
Great minds think alike: https://github.com/Martin2112/trillian/tree/rpc_server_healthcheck. You've done the kubernetes config as well though.

File: server/main.go [hunk 1]
```diff
@@ -80,6 +85,18 @@ type Main struct {
 	TreeDeleteMinInterval time.Duration
 }
 
+func (m *Main) healthz(rw http.ResponseWriter, req *http.Request) {
+	if m.IsHealthy != nil {
+		if err := m.IsHealthy(req.Context()); err != nil {
```
Comment (Martin2112) [inline server/main.go]: I think (could be wrong) that the request context defaults to context.Background() so might need an explicit deadline here.

File: examples/deployment/kubernetes/trillian-map-deployment.yaml [hunk 2]
```diff
@@ -31,7 +31,7 @@ spec:
             command:
             - curl
             - -f
-            - http://localhost:8091/metrics
+            - http://localhost:8091/healthz
           failureThreshold: 3
           periodSeconds: 30
           timeoutSeconds: 30
```
Comment (Martin2112) [inline examples/deployment/kubernetes/trillian-map-deployment.yaml]: 30 seconds is prob. too long given we only have 10? sec for live requests.

File: server/main.go [hunk 3]
```diff
@@ -80,6 +85,18 @@ type Main struct {
...
+			rw.WriteHeader(http.StatusServiceUnavailable)
+			rw.Write([]byte(err.Error()))
+			return
+		}
+	}
+	rw.WriteHeader(http.StatusOK)
```
Comment (Martin2112) [inline server/main.go]: Pretty sure there's an implicit 200 status if not overridden.

File: examples/deployment/kubernetes/trillian-map-deployment.yaml (hunk 2 above)
Comment (AlCutter) [reply examples/deployment/kubernetes/trillian-map-deployment.yaml]: done

File: server/main.go (hunk 3 above)
Comment (AlCutter) [reply server/main.go]: done

File: server/main.go (hunk 1 above)
Comment (AlCutter) [reply server/main.go]: Done

**Pull request merged**